test/streams/broadcastable_test.rb

```diff
@@ -36,6 +36,12 @@ class Turbo::BroadcastableTest < ActionCable::Channel::TestCase
     end
   end
 
+  test "broadcasting update to stream now with template option" do
+    assert_broadcast_on "stream", turbo_stream_action_tag("update", target: "message_1", template: render("messages/index", layout: false)) do
+      @message.broadcast_update_to "stream", template: "messages/index"
+    end
+  end
```

Comment:
I'm wondering if we'd want:

* Only one method tested with the `template` option, like the current commit has.
* Separate tests for each `Broadcastable` method. e.g.

```rb
  test "broadcasting update to stream now with template option" do
    assert_broadcast_on "stream", turbo_stream_action_tag("update", target: "message_1", template: render("messages/index", layout: false)) do
      @message.broadcast_update_to "stream", template: "messages/index"
    end
  end

  test "broadcasting replace to stream now with template option" do
    assert_broadcast_on "stream", turbo_stream_action_tag("replace", target: "message_1", template: render("messages/index", layout: false)) do
      @message.broadcast_replace_to "stream", template: "messages/index"
    end
  end
```

* Separate methods in a dynamic way, e.g.:

```rb 
  [
    [:broadcast_update_to, :update],
    [:broadcast_replace_to, :replace]
  ].each do |method, action|
    test "broadcasting #{method} to stream now with template option" do
      assert_broadcast_on "stream", turbo_stream_action_tag(action, target: "message_1", template: render("messages/index", layout: false)) do
        @message.public_send method, "stream", template: "messages/index"
      end
    end
  end

  [
    [:broadcast_before_to, :before],
    [:broadcast_after_to, :after],
    [:broadcast_append_to, :append],
    [:broadcast_prepend_to, :prepend],
  ].each do |method, action|
    test "broadcasting #{method} to stream now with template option" do
      assert_broadcast_on "stream", turbo_stream_action_tag(action, target: "message_1", template: render("messages/index", layout: false)) do
        @message.public_send method, "stream", template: "messages/index", target: "message_1"
      end
    end
  end

  # ...etc.
```

Reason I only added one test for now is because the current tests are not thoroughly testing all possible rendering options for all methods. e.g. the `partial` option is only tested in the test named `local variables don't get overwritten if they collide with the template name`, so I don't know how to proceed.

Reply:
Prefer the separate tests. Please do open a follow up PR to expand test coverage!